### PR TITLE
Add option to disable ssl verification

### PR DESCRIPTION
**Pull request by Mys7ic**
In some cases the ssl verification fails

```
SSL_connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed (OpenSSL::SSL::Error)
```
in some situations it is useful to deactivate verification

### CODE REVIEW

File: src/wafalyzer.cr [hunk 1]
```diff
@@ -21,6 +21,7 @@ module Wafalyzer
       port: port,
       tls: tls,
     )
+    client.tls.verify_mode=OpenSSL::SSL::VerifyMode::NONE if settings.disable_ssl_verifications?
```
Comment (Sija) [inline src/wafalyzer.cr]:
```suggestion
    client.tls.verify_mode = :none if settings.disable_ssl_verification?
```

File: src/cli.cr [hunk 2]
```diff
@@ -28,6 +28,7 @@ timeout = nil
 fallback_requests_count = nil
 json = false
 use_random_user_agent = false
+disable_ssl_verifications = false
```
Comment (Sija) [inline src/cli.cr]:
```suggestion
disable_ssl_verification = false
```

File: src/cli.cr [hunk 3]
```diff
@@ -66,6 +67,10 @@ OptionParser.parse do |parser|
     use_random_user_agent = true
   end
 
+  parser.on "-s", "--disable-ssl-verify", "Disable SSL verifications" do
+    disable_ssl_verifications = true
+  end
```
Comment (Sija) [inline src/cli.cr]:
```suggestion
  parser.on "-s", "--disable-ssl-verify", "Disable SSL verification" do
    disable_ssl_verification = true
  end
```

File: src/cli.cr [hunk 4]
```diff
@@ -94,6 +99,7 @@ Wafalyzer.configure do |settings|
   settings.timeout = timeout
   settings.fallback_requests_count = fallback_requests_count
   settings.use_random_user_agent = use_random_user_agent
+  settings.disable_ssl_verifications = disable_ssl_verifications
```
Comment (Sija) [inline src/cli.cr]:
```suggestion
  settings.disable_ssl_verification = disable_ssl_verification
```

File: src/wafalyzer/settings.cr [hunk 5]
```diff
@@ -27,6 +27,9 @@ module Wafalyzer
     # `User-Agent` string from the `user_agents` array.
     class_property? use_random_user_agent = false
 
+    # Setting it to `true` will disable SSL verifications
+    class_property? disable_ssl_verifications = false
```
Comment (Sija) [inline src/wafalyzer/settings.cr:31]:
```suggestion
    class_property? disable_ssl_verification = false
```

**Review by Sija — CHANGES_REQUESTED** [review]

File: src/wafalyzer/settings.cr (hunk 5 above)
Comment (Sija) [inline src/wafalyzer/settings.cr]:
```suggestion
    # Setting it to `true` will disable SSL verification
    class_property? disable_ssl_verification = false
```

**Comment by Sija** [comment]
I wonder, wouldn't be better to provide an option allowing to set the desired `verify_mode` to the given [option](https://crystal-lang.org/api/1.1.0/OpenSSL/SSL/VerifyMode.html)?

/cc @bararchy

**Review by Sija — APPROVED** [review]
Other than https://github.com/NeuraLegion/wafalyzer/pull/5#issuecomment-883453903 it's GTG from my side

**Comment by FLX-0x00** [comment]
Set VerifyMode as options "array" as the other options in wafalyzer might be a good idea to finetune. Most cases "none" should be "fix" the most cases. As the application exit with a non controlled error this will be a nice hotfix. In the future it will might be nice to control the error. But maybe the other options are too much "detailed"

```
CLIENT_ONCE = 4
FAIL_IF_NO_PEER_CERT = 2
NONE = 0
PEER = 1
```

File: src/wafalyzer.cr [hunk 6]
```diff
@@ -21,6 +21,7 @@ module Wafalyzer
       port: port,
       tls: tls,
     )
+    client.tls.verify_mode = :none if settings.disable_ssl_verification?
```
Comment (Sija) [inline src/wafalyzer.cr]: I forgot about the case in which `client.tls` might be `nil`.

```suggestion
    client.tls?.try(&.verify_mode = :none) if settings.disable_ssl_verification?
```

**Review by Sija — CHANGES_REQUESTED** [review]

**Review by Sija — APPROVED** [review]

File: src/cli.cr [hunk 7]
```diff
@@ -66,6 +67,10 @@ OptionParser.parse do |parser|
     use_random_user_agent = true
   end
 
+  parser.on "-s", "--disable-ssl-verify", "Disable SSL verification" do
```
Comment (Sija) [inline src/cli.cr]:
```suggestion
  parser.on "-s", "--disable-ssl-verify", "Disables SSL verification" do
```

**Pull request merged**